sonarts-sq-plugin/sonar-typescript-plugin/src/main/java/org/sonar/plugin/typescript/SensorContextUtils.java

```diff
@@ -0,0 +1,269 @@
+/*
+ * SonarTS
+ * Copyright (C) 2017-2017 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.plugin.typescript;
+
+import com.google.gson.JsonElement;
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import org.sonar.api.batch.fs.FilePredicate;
+import org.sonar.api.batch.fs.FileSystem;
+import org.sonar.api.batch.fs.InputFile;
+import org.sonar.api.batch.sensor.SensorContext;
+import org.sonar.api.batch.sensor.issue.NewIssue;
+import org.sonar.api.batch.sensor.issue.NewIssueLocation;
+import org.sonar.api.rule.RuleKey;
+import org.sonar.api.utils.log.Logger;
+import org.sonar.api.utils.log.Loggers;
+
+public class SensorContextUtils {
+
+  private static final Logger LOG = Loggers.get(ContextualSensor.class);
+
+  private SensorContextUtils() {
+  }
+
+  static void saveIssues(SensorContext sensorContext, Issue[] issues, TypeScriptRules typeScriptRules) {
+    FileSystem fs = sensorContext.fileSystem();
+    for (Issue issue : issues) {
+      InputFile inputFile = fs.inputFile(fs.predicates().hasAbsolutePath(issue.name));
+      if (inputFile != null) {
+        saveIssue(sensorContext, typeScriptRules, issue, inputFile);
+      }
+    }
+  }
+
+  public static List<RuleToExecute> convertToRulesToExecute(TypeScriptRules typeScriptRules) {
+    List<RuleToExecute> rulesToExecute = new ArrayList<>();
+
+    typeScriptRules.forEach(rule -> {
+      if(rule.isEnabled()) {
+        rulesToExecute.add(new RuleToExecute(rule.tsLintKey(), rule.configuration()));
+      }
+    });
+
+    return rulesToExecute;
+  }
+
+  static void saveIssuePocToRemove(SensorContext sensorContext, TypeScriptRules typeScriptRules, Issue issue, InputFile inputFile) {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Remove this unused method parameter "typeScriptRules". ([squid:S1172](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1172))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=sonarts&branch=sonarlint-integration-poc&issues=AWDgL5jfbBlXo5JPxg67&open=AWDgL5jfbBlXo5JPxg67)

[ISSUE_KEY]: AWDgL5jfbBlXo5JPxg67